nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -4,19 +4,45 @@
 import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.tree.AnnotatedTypeTree;
 import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.AssignmentTree;
+import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.ParameterizedTypeTree;
 import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
 import com.sun.tools.javac.code.Attribute;
 import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 
 /** Methods for performing checks related to generic types and nullability. */
 public final class GenericsChecks {
+  VisitorState state;
+  Config config;
+  NullAway analysis;
 
-  private GenericsChecks() {
-    // just utility methods
+  public GenericsChecks(VisitorState state, Config config, NullAway analysis) {
+    this.state = state;
+    this.config = config;
+    this.analysis = analysis;
+  }
+
+  @SuppressWarnings("UnusedVariable")
+  public static Type supertypeMatchingLHS(
+      Type.ClassType lhsType, Type.ClassType rhsType, VisitorState state) {
+    // all supertypes including classes as well as interfaces
+    List<Type> listOfDirectSuperTypes = state.getTypes().closure(rhsType);
+    if (listOfDirectSuperTypes != null) {
```

Comment:
instead of checking if it is a direct supertype first and then calling this function I am just calling the function and checking it here. I am not sure what is the logic behind checking whether it is the direct super type or not. considering it iterates through the list of the super types and checks whether it is a direct supertype and then again checks in the same list for the direct supertype matching the lhs type, it will check the same thing twice.